src/test/kotlin/CrashReaderIntegrationTest.kt

```diff
@@ -1,85 +1,108 @@
 import arrow.core.Either
 import me.urielsalis.mccrashlib.Crash
 import me.urielsalis.mccrashlib.CrashReader
-import me.urielsalis.mccrashlib.parser.ParserError
-import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.ArgumentsProvider
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.junit.jupiter.params.support.AnnotationConsumer
 import java.io.File
 import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Stream
+import kotlin.streams.toList
 
 class CrashReaderIntegrationTest {
-    val crashReader = CrashReader()
-    val tempDir = Files.createTempDirectory("mappings").toFile()
+    @Target(AnnotationTarget.FUNCTION)
+    @Retention(AnnotationRetention.RUNTIME)
+    @ArgumentsSource(ResourceFilesProvider::class)
+    annotation class ResourceFilesSource(val directory: String, val ignoredNameSuffixes: Array<String> = [])
```

Comment:
I have added a custom arguments provider here to simplify the test methods; the code is roughly based on JUnit 5's [`MethodArgumentsProvider`](https://github.com/junit-team/junit5/blob/main/junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/MethodArgumentsProvider.java).

Please let me know if you don't think this amount of boilerplate code is worth it.